tests/qos/test_buffer_traditional.py

```diff
@@ -0,0 +1,222 @@
+
+            if pg_name_map:
+                for pg in default_lossless_pgs:
+                    buffer_pg_asic_oid = pg_name_map['{}:{}'.format(port, pg)]
+                    buffer_pg_asic_key = duthost.shell('redis-cli -n 1 keys *{}*'.format(buffer_pg_asic_oid))['stdout']
+                    buffer_profile_oid_in_pg = duthost.shell('redis-cli -n 1 hget {} SAI_INGRESS_PRIORITY_GROUP_ATTR_BUFFER_PROFILE'.format(buffer_pg_asic_key))['stdout']
+                    logging.info("Checking admin-up port {} lossless PG {} in ASIC_DB ({})".format(port, pg, buffer_profile_oid_in_pg))
+                    if buffer_profile_oid:
+                        if not _check_condition(buffer_profile_oid == buffer_profile_oid_in_pg,
+                                                "Different OIDs in PG 3 ({}) and 4 ({}) in port {}".format(buffer_profile_oid, buffer_profile_oid_in_pg, port),
+                                                use_assert):
+                            return None, False
+                    else:
+                        buffer_profile_oid = buffer_profile_oid_in_pg
+        else:
+            if not _check_condition(not profile_in_pg, "Buffer PG configured on admin down port {}".format(port), use_assert):
+                return None, False
+            if pg_name_map:
+                for pg in default_lossless_pgs:
+                    buffer_pg_asic_oid = pg_name_map['{}:{}'.format(port, pg)]
+                    buffer_pg_asic_key = duthost.shell('redis-cli -n 1 keys *{}*'.format(buffer_pg_asic_oid))['stdout']
+                    buffer_profile_oid_in_pg = duthost.shell('redis-cli -n 1 hget {} SAI_INGRESS_PRIORITY_GROUP_ATTR_BUFFER_PROFILE'.format(buffer_pg_asic_key))['stdout']
+                    logging.info("Checking admin-down port {} lossless PG {}".format(port, pg))
+                    if not _check_condition(not buffer_profile_oid_in_pg or buffer_profile_oid_in_pg == 'oid:0x0',
+                                            "Buffer PG configured on admin down port in ASIC_DB {}".format(port),
+                                            use_assert):
+                        return None, False
+
+        return buffer_profile_oid, True
+
+    def _check_port_buffer_info_and_return(duthost, port, expected_profile):
+        """Check port's buffer information against CONFIG_DB and ASIC_DB and return the result
+
+        This is called from wait_until
+
+        Args:
+            duthost: The duthost object
+            port: The port to test in string
+            expected_profile: The expected profile in string
+
+        Return:
+            Whether all the checks passed
+        """
+        _, result = _check_port_buffer_info_and_get_profile_oid(duthost, port, expected_profile, False)
+        return result
+
+    duthost = duthosts[rand_one_dut_hostname]
+    if duthost.facts["asic_type"] != "mellanox":
+        pytest.skip("Traditional buffer test runs on Mellanox platform only, skip")
```

Comment:
Please move this to an autouse module scoped fixture

Reply:
Fixed